public/app/features/dashboard/components/DashboardPermissions/DashboardPermissions.tsx

```diff
@@ -1,34 +1,42 @@
 import React, { PureComponent } from 'react';
-import { Tooltip, Icon } from '@grafana/ui';
+import { connect, MapStateToProps, MapDispatchToProps } from 'react-redux';
+import { Tooltip, Icon, Button } from '@grafana/ui';
 import { SlideDown } from 'app/core/components/Animations/SlideDown';
-import { StoreState, FolderInfo } from 'app/types';
+import { StoreState } from 'app/types';
 import { DashboardAcl, PermissionLevel, NewDashboardAclItem } from 'app/types/acl';
 import {
   getDashboardPermissions,
   addDashboardPermission,
   removeDashboardPermission,
   updateDashboardPermission,
 } from '../../state/actions';
+import { DashboardModel } from '../../state/DashboardModel';
 import PermissionList from 'app/core/components/PermissionList/PermissionList';
 import AddPermission from 'app/core/components/PermissionList/AddPermission';
 import PermissionsInfo from 'app/core/components/PermissionList/PermissionsInfo';
-import { connectWithStore } from 'app/core/utils/connectWithReduxStore';
 
-export interface Props {
-  dashboardId: number;
-  folder?: FolderInfo;
+export interface OwnProps {
+  dashboard: DashboardModel;
+}
+
+export interface ConnectedProps {
   permissions: DashboardAcl[];
+}
+
+export interface DispatchProps {
   getDashboardPermissions: typeof getDashboardPermissions;
   updateDashboardPermission: typeof updateDashboardPermission;
   removeDashboardPermission: typeof removeDashboardPermission;
   addDashboardPermission: typeof addDashboardPermission;
 }
 
+export type Props = OwnProps & ConnectedProps & DispatchProps;
+
```

Comment:
As you added Own & ConnectedProps maybe you should follow the guide here too: https://github.com/grafana/grafana/blob/master/contribute/style-guides/redux.md#typing-of-connected-props

Reply:
Thanks for the link @hugohaggmark. I've updated this so types are automatically inferred! ⚡

Reply:
Awesome!